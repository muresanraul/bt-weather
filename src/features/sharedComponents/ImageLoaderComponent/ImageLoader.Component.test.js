import React from 'react';
import renderer from 'react-test-renderer';

import { default as ImageLoaderComponent } from './ImageLoader.Component';

describe('ImageLoaderComponent:', () => {

  it('Component - renders correctly with no properties', () => {
    const imageLoaderComponent = renderer.create(
      <ImageLoaderComponent />
    ).toJSON();
    expect(imageLoaderComponent).toMatchSnapshot();
  });

  it('Component - renders correctly with set image', () => {
    const props = {img: '/img.jpg'};
    const imageLoaderComponent = renderer.create(
      <ImageLoaderComponent {...props}/>
    ).toJSON();
    expect(imageLoaderComponent).toMatchSnapshot();
  });

  it('Component - renders correctly with set image and alt', () => {
    const props = {img: '/img.jpg', alt: 'alt'};
    const imageLoaderComponent = renderer.create(
      <ImageLoaderComponent {...props}/>
    ).toJSON();
    expect(imageLoaderComponent).toMatchSnapshot();
  });

  it('Component - renders correctly with no image if the alt property is set but the img prop is not set', () => {
    const props = {alt: 'alt'};
    const imageLoaderComponent = renderer.create(
      <ImageLoaderComponent {...props}/>
    ).toJSON();
    expect(imageLoaderComponent).toMatchSnapshot();
  });

});
