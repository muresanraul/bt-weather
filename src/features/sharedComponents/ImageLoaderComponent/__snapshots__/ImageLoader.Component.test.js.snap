// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageLoaderComponent: Component - renders correctly with no image if the alt property is set but the img prop is not set 1`] = `
<div
  className="image-loader"
>
  <div
    className="image-loader__placeholder"
  />
</div>
`;

exports[`ImageLoaderComponent: Component - renders correctly with no properties 1`] = `
<div
  className="image-loader"
>
  <div
    className="image-loader__placeholder"
  />
</div>
`;

exports[`ImageLoaderComponent: Component - renders correctly with set image 1`] = `
<div
  className="image-loader"
>
  <img
    alt=""
    src="http://openweathermap.org/img/wn//img.jpg@2x.png"
  />
</div>
`;

exports[`ImageLoaderComponent: Component - renders correctly with set image and alt 1`] = `
<div
  className="image-loader"
>
  <img
    alt="alt"
    src="http://openweathermap.org/img/wn//img.jpg@2x.png"
  />
</div>
`;
